lib/rucio/core/replica.py

```diff
@@ -1600,7 +1601,22 @@ def _expected_pfns(lfns, rse_settings, scheme, operation='write', domain='wan',
 
 
 @transactional_session
-def add_replica(rse_id, scope, name, bytes_, account, adler32=None, md5=None, dsn=None, pfn=None, meta=None, rules=[], tombstone=None, *, session: "Session"):
+def add_replica(
+    rse_id: str,
+    scope: InternalScope,
+    name: str,
+    bytes_: int,
+    account: models.InternalAccount,
+    adler32: "Optional[str]" = None,
+    md5: "Optional[str]" = None,
+    dsn: "Optional[str]" = None,
+    pfn: "Optional[str]" = None,
+    meta: "Optional[dict[str, Any]]" = None,
+    rules: "Optional[list[dict[str, Any]]]" = None,
+    tombstone: "Optional[datetime]" = None,
+    *,
+    session: "Session"
+) -> bool:
```

Comment:
This actually returns a `list[dict]`. 

`add_replica` returns `add_replicas`: https://github.com/rucio/rucio/blob/19392c29c61021d37aa1f3bbe6017722222994cf/lib/rucio/core/replica.py#L1644

Which returns `replicas`: https://github.com/rucio/rucio/blob/19392c29c61021d37aa1f3bbe6017722222994cf/lib/rucio/core/replica.py#L1600

Which is defined here: https://github.com/rucio/rucio/blob/19392c29c61021d37aa1f3bbe6017722222994cf/lib/rucio/core/replica.py#L1562

Which returns this list:

https://github.com/rucio/rucio/blob/19392c29c61021d37aa1f3bbe6017722222994cf/lib/rucio/core/replica.py#L1453

So it's a case of the existing docstrings being wrong, as they say that all these functions return a `bool`, which is not the case